crates/proof-of-sql/src/utils/parse.rs

```diff
@@ -0,0 +1,91 @@
+use crate::base::map::IndexMap;
+use alloc::string::ToString;
+use sqlparser::{
+    ast::{DataType, ExactNumberInfo, Statement},
+    dialect::GenericDialect,
+    parser::Parser,
+};
+
+/// Parse a DDL file and return a map of table names to bigdecimal columns
+///
+/// # Panics
+/// Panics if there is an error parsing the SQL
+#[must_use]
+pub fn find_bigdecimals(queries: &str) -> IndexMap<String, Vec<String>> {
```

Comment:
I don't know for sure if this is necessary, but I will also need precision and scale for the decimals.

Reply:
@stuarttimwhite Ah. Since we actually have decimals with precisions beyond Decimal256 what about returning `IndexMap<String, Vec<(u8, i8)>>` instead?